### ISSUE TITLE
does not install on my windows 10

### ISSUE THREAD

**Issue by rfikki** [OpenZeppelin/zeppelin-solidity#96]
$ npm i zeppelin-solidity

> zeppelin-solidity@1.0.0 install C:\projects\truffle-dapps\coinmaker\node_modul                                   es\zeppelin-solidity
> scripts/install.sh

'scripts' is not recognized as an internal or external command,
operable program or batch file.
npm WARN enoent ENOENT: no such file or directory, open 'C:\projects\truffle-dap                                   ps\coinmaker\package.json'
npm WARN coinmaker No description
npm WARN coinmaker No repository field.
npm WARN coinmaker No README data
npm WARN coinmaker No license field.
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Rocky\\AppData\\                                   Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "i" "zeppelin-solidity"
npm ERR! node v6.7.0
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE

npm ERR! zeppelin-solidity@1.0.0 install: `scripts/install.sh`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the zeppelin-solidity@1.0.0 install script 'scripts/install.s                                   h'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the zeppelin-solidity pac                                   kage,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     scripts/install.sh
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs zeppelin-solidity
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls zeppelin-solidity
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\projects\truffle-dapps\coinmaker\npm-debug.log

Rocky@ASUS MINGW64 /c/projects/truffle-dapps/coinmaker
$ node --version
v6.7.0

Rocky@ASUS MINGW64 /c/projects/truffle-dapps/coinmaker
$ npm --version
3.10.8

**Comment by maraoz** [comment]
@rfikki thanks for reporting. Seems like a problem with how windows handles paths.

Can you try changing this line: https://github.com/OpenZeppelin/zeppelin-solidity/blob/master/package.json#L12 
to something like: 
```
"install": "./scripts/install.sh"
```

Or however windows recognizes the path? (sorry, I don't have much experience with windows development)
Let me know

**Comment by maraoz** [comment]
@rfikki any update on this? Would like to help

**Comment by DiscRiskandBisque** [comment]
@maraoz I added that as well as tried added exec in front of the path:
`exec ./scripts/install.sh`

Neither worked unfortunately. I've been researching other ways of moving files post-npm-install to try and create a helpful pull request, but am not having much luck yet.

**Comment by explodingcamera** [comment]
You can't run bash (Unix/Linux) scripts on windows by default if you're not running something like wsl (Windows Subsystem for Linux) or GIT Bash. I think using os-specific scripts/commands and copying stuff to some folder outside of node_modules is a bad Idea, you should convert your bash scripts to js files and add the moving of folders as part of a global commad, for example `zeppelin init` if users want to use that.

I'm currently running on windows so I could do a pr if you want.

As temp workaround can also use `npm install zeppelin-solidity --ignore-scripts` to skip the post-install script.

**Comment by maraoz** [comment]
@explodingcamera Thanks for the info.
The post-install script should be gone once we move to ethpm, so that should also fix this problem.
No need to do what you suggested, in my opinion :)

**Comment by ProphetDaniel** [comment]
To get rid of the errors we were having before we started powershell as admin and run the following command line:

`npm  install zeppelin-solidity --ignore-scripts`

The result was:

```
`-- zeppelin-solidity@1.0.4
  `-- truffle-hdwallet-provider@0.0.3
    +-- web3@0.18.4
    | `-- bignumber.js@2.0.7  (git+https://github.com/debris/bignumber.js.git#94d7146671b9719e00a09c29b01a691bc85048c2)
    `-- web3-provider-engine@8.6.1
      `-- web3@0.16.0
        `-- bignumber.js@2.0.7  (git+https://github.com/debris/bignumber.js.git#94d7146671b9719e00a09c29b01a691bc85048c2)

npm WARN enoent ENOENT: no such file or directory, open 'C:\WINDOWS\system32\package.json'
npm WARN system32 No description
npm WARN system32 No repository field.
npm WARN system32 No README data
npm WARN system32 No license field.
```
 
To get rid of the warnings we added a package.json file with the following content to WINDOWS\System32:

```
{
  "description": "Holding package.json",
  "repository": {
    "type": "git",
    "url": "A repository path"
  },
  "license": "MIT"
}

```

Finally the open zeppelin package installed without any errors or warnings. To be tested now though.

```
C:\WINDOWS\system32> npm install zeppelin-solidity --ignore-scripts
C:\WINDOWS\system32
`-- zeppelin-solidity@1.0.4
  `-- truffle-hdwallet-provider@0.0.3
    +-- web3@0.18.4
    | `-- bignumber.js@2.0.7  (git+https://github.com/debris/bignumber.js.git#94d7146671b9719e00a09c29b01a691bc85048c2)
    `-- web3-provider-engine@8.6.1
      `-- web3@0.16.0
        `-- bignumber.js@2.0.7  (git+https://github.com/debris/bignumber.js.git#94d7146671b9719e00a09c29b01a691bc85048c2)
```

**Comment by ProphetDaniel** [comment]
After issueing the `truffle install zeppelin` command just after `truffle init` inside a new folder:
Unfortunately the following error is raised:

```
Secp256k1 bindings are not compiled. Pure JS implementation will be used.
Error: Could not find object at hash 'QmYt6ZfD1VNmKp85TeteEcsgaxHzP2gTSKupHxHMVgcHYb' in 5000ms
    at Timeout._onTimeout (C:\Users\user\AppData\Roaming\npm\node_modules\truffle\node_modules\ethpm\lib\hosts\ipfshost.
js:47:14)
    at ontimeout (timers.js:386:14)
    at tryOnTimeout (timers.js:250:5)
    at Timer.listOnTimeout (timers.js:214:5)
```

**Comment by maraoz** [comment]
This is a known issue with IPFS, sorry: https://github.com/OpenZeppelin/zeppelin-solidity/issues/209
We're working on fixing it

**Comment by rstormsf** [comment]
this issue could be fixed when #253 is deployed to npm, correct?

**Comment by frangio** [comment]
This is obsolete as `scripts/install.sh` was removed. If there are any newer installation problems on Windows please open a new issue.
